src/core_plugins/kibana/public/management/react/index-pattern/index-pattern-view/index-pattern-view.component.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import {
+  KuiPageContent,
+  KuiPageContentBody,
+  KuiPageContentHeader,
+  KuiPageContentHeaderSection,
+  KuiTitle,
+  KuiText,
+  KuiButtonIcon,
+  KuiTabs,
+  KuiTab,
+  KuiButtonEmpty,
+  KuiFlexGroup,
+  KuiFlexItem,
+  KuiConfirmModal,
+  KuiModalOverlay,
+} from 'ui_framework/components';
+
+import { IndexPatternFields } from './components/index-pattern-fields';
+
+const IndexPatternView = ({
+  pattern,
+  id,
+  fields,
+  timeFieldName,
+  isDefault,
+  selectedTab,
+  changeTab,
+  isShowingRefreshFieldsConfirmation,
+  showRefreshFieldsConfirmation,
+  hideRefreshFieldsConfirmation,
+  refreshFields,
+  deleteIndexPattern,
+  setDefaultIndexPattern,
+}) => {
+  if (pattern === undefined || fields === undefined) {
+    return null;
+  }
+
+  let tabContent = null;
+  if (selectedTab === 'fields') {
+    tabContent = <IndexPatternFields fields={fields}/>;
+  }
+
+  return (
+    <KuiPageContent>
+      <KuiPageContentHeader>
+        <KuiPageContentHeaderSection>
+          <KuiFlexGroup>
+            <KuiFlexItem>
+              <KuiTitle>
+                <h2>{pattern}</h2>
+              </KuiTitle>
+            </KuiFlexItem>
+            { !isDefault
+              ?
+                <KuiFlexItem>
+                  <KuiButtonEmpty onClick={() => setDefaultIndexPattern(id)}>
+                    Set as default index
+                  </KuiButtonEmpty>
+                </KuiFlexItem>
+              : null
+            }
+          </KuiFlexGroup>
+          { isShowingRefreshFieldsConfirmation
+            ?
+              <KuiModalOverlay>
```

Comment:
This could be a good opportunity for extracting code into a new component, e.g. `RefreshFieldsModal`, possibly with its own container so this component won't need to know about `hideRefreshFieldsConfirmation` or `refreshFields`. This will simplify this component and allow us to test the modal independently.